docs/architecture.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: default
+title: Architecture
+nav_order: 1
+---
+
+## Architecture
+
+![architecture_diagram.png](architecture_diagram.png)
+
+[Architecture diagram, full scale](architecture_diagram.png)
+
+
+## Data flow
```

Comment:
This section is mentioned and linked in the 'Data flow of the notification service without cooldown' section of the `index.md` file. Please update .